### DISCUSSION TITLE
Using Eventuous to build aggregates using server projections like $ce-

### DISCUSSION THREAD

**Discussion by peteglassey** [Eventuous/eventuous#326] [General]
I want to use the IEventStore or IEventReader to build up an aggregate based on an ESDB server projected stream - in my case a category `$ce-xyz`. I found that i always get no events back when using the ReadEvents() extension method on StoreFunctions. When i debug through the ESDB EventStore Eventuous is explicitly filtering out all events from streams that start with $.

Is there another way instead of having to fall back to using the ESDB native SDK and reproducing all the logic in Eventuous in my own source? Is there another layer of abstraction i could make use of? 

I know it sounds like a slightly odd thing to do, but i want to load all the events (there is likely < 2 or 3 events per stream, and maybe 2 or 3 dozen streams) by category into memory and do some in memory grouping  and build up aggregates from that. 

How do we build aggregates from a server-side projection using Eventuous?

As a stop-gap, i have a read-model projection which builds what i need quite simply, but i think i would rather prefer to build the state i need from an event stream than get it from a read-model projection.

**Comment by alexeyzimarev** [comment]
I would strongly suggest avoiding `$ce` and links in general. Since v25.1 the secondary indexes feature does a lot better job than links (which cannot be deleted). In both cases, the library expects an aggregate to have its own stream. For other cases I can only suggest building a custom implementation of event store. It's a simple interface.
